src/freenet/node/NodeIPDetector.java

```diff
@@ -526,6 +530,27 @@ public void set(String val) throws InvalidConfigValueException {
 				oldIPAddress = null;
 			}
 		}
+
+		// allow binding to localhost, 127.0.0.1, ...?
+
+		nodeConfig.register("allowBindToLocalhost", false, sortOrder++, true, false, "Node.allowBindToLocalhost", "Node.allowBindToLocalhostLong", new BooleanCallback() {
+
+			@Override
+			public Boolean get() {
+				return allowBindToLocalhost;
+			}
+
+			@Override
+			public void set(Boolean val) throws NodeNeedRestartException {
+				if (allowBindToLocalhost != val) {
+					synchronized(this) {
```

Comment:
This synchronization is broken, as long as there are places where `allowBindToLocalhost` is accessed without synchronization, such as in the `get()` method.